core/coreunix/add.go

```diff
@@ -108,19 +100,28 @@ type Adder struct {
 	NoCopy     bool
 	Chunker    string
 	root       node.Node
-	mr         *mfs.Root
+	mroot      *mfs.Root
 	unlocker   bs.Unlocker
 	tempRoot   *cid.Cid
-	prefix     cid.Prefix
+	Prefix     cid.Prefix
 }
 
-func (adder *Adder) SetPrefix(prefix cid.Prefix) {
-	adder.mr.SetPrefix(prefix)
-	adder.prefix = prefix
+func (adder *Adder) mfsRoot() (*mfs.Root, error) {
+	if adder.mroot != nil {
+		return adder.mroot, nil
+	}
+	rnode := unixfs.EmptyDirNode()
+	rnode.SetPrefix(adder.Prefix)
+	mr, err := mfs.NewRoot(adder.ctx, adder.dagService, rnode, nil)
+	if err != nil {
+		return nil, err
+	}
+	adder.mroot = mr
+	return adder.mroot, nil
 }
 
 func (adder *Adder) SetMfsRoot(r *mfs.Root) {
-	adder.mr = r
+	adder.mroot = r
```

Comment:
I can change it back.  I did this as a trick to make sure I cached where all instance `mr` was used.  I thought about naming the method `mr()` but that seamed a bit terse for a method name, even if it was only for internal use.